src/lavatory/utils/artifactory.py

```diff
@@ -1,100 +1,90 @@
+                repos[repo["repoKey"]] = repo
 
-        """
-        response = requests.delete(baseurl + "/" + path)
+        return repos
 
-        if response.ok:
-            return response.json()
-        else:
-            response.raise_for_status()
-            #raise Exception("HTTP DELETE {}: Server code {}".format(endpoint, response.status_code))
-        """
+    def all_artifacts(self, search='', repo='', depth=3):
+        """ Returns a dict of artifact and properties """
+        LOG.debug('Finding all artifacts with: search=%s, repo=%s, depth=%s', search, repo, depth)
+        all_artifacts = {}
 
+        self.artifactory.find_by_pattern(filename=search, specific_repo=repo, max_depth=depth)
 
-class Artifactory(object):
+        for artifact in sorted(self.artifactory.files):
+            artifact_simple_name = self._parse_artifact_name(artifact)
+            LOG.debug('Found: %s', artifact_simple_name)
+            self.artifactory.get_properties(artifact)
 
-    def __init__(self, baseurl, user, passwd):
-        self.http = _Http(baseurl, user, passwd)
+        LOG.info('Found %d artifacts in total.', len(all_artifacts))
+        return all_artifacts
 
-    def list(self, reponame=None):
-        """
-        Return a list of repos with basic info about each
-        If the optional parameter reponame is specified, then only return
-        information pertaining to that repo
-        """
-    
-        repos = {}
-    
-        json = self.http.get("/storageinfo")
-        for repo in json["repositoriesSummaryList"]:
-            if repo["repoKey"] == "TOTAL" :
-                continue
-                
-            if not reponame or reponame == repo["repoKey"]:
-                repos[repo["repoKey"]] = repo
-    
-        return repos
-    
-    def purge(self, repo, dryrun, artifacts):
+    def purge(self, repo, dry_run, artifacts):
         """ Purge artifacts from the specified repo
 
         Keyword arguments:
         repo -- the repo to target for this operation
-        dryrun -- false to execute an actual purge
+        dry_run -- false to execute an actual purge
         artifacts -- a list of artifacts to operate upon
         """
-
         purged = 0
-        mode = "DRYRUN" if dryrun else "LIVE"
+        mode = "DRYRUN" if dry_run else "LIVE"
 
         for artifact in artifacts:
             LOG.info("  {} purge {}:{}".format(mode, repo, artifact))
-            if dryrun:
+            if dry_run:
                 purged += 1
             else:
                 try:
-                    self.http.delete(artifact)
+                    self.artifactory.request(artifact, method='delete')
                     purged += 1
                 except Exception as e:
                     LOG.error(str(e))
 
         return purged
 
-    def filter(self, repo, terms=[], depth=3, sort=None, offset=None, limit=None):
+    def filter(self, repo, terms=None, depth=3, sort=None, offset=None, limit=None):
```

Comment:
repo can be removed from this function and use`self.repo_name`

Reply:
changed to repo_name from init.